File: CWE_Java/src/main/java/com/example/cwe/CWE306/NonComplaint/NonComplaint.java
package com.example.cwe.CWE306.NonComplaint;

//      CWE-306: Missing Authentication for Critical Function

public class NonComplaint {
    public static void main(String[] args) {
        BankAccount user1 = createBankAccount("123","saving","Owner",300);
        user1.getOwnerName();
    }

    public static BankAccount createBankAccount(String accountNumber, String accountType,
                                                String accountOwnerName, double balance) {
        BankAccount account = new BankAccount();
        account.setAccountNumber(accountNumber);
        account.setAccountType(accountType);
        account.setAccountOwnerName(accountOwnerName);
        account.setBalance(balance);

        return account;
    }

}

File: CWE_JavaScipt/src/CWE306/NonComplaint.js
//      CWE-306	Missing Authentication for Critical Function

class User {

    setAccountNumber(accNo) {
            this.accountNumber = accNo;
        }

    setAccountType(accType) {
            this.accountType = accType;
        }

    setAccountOwnerName(accName) {
            this.accountName = accName;
        }

    setBalance(bal) {
            this.balance = bal;
        }

    getAccountDetail(){
    console.log("AccountNumber:", this.accountNumber)
    console.log("Owner Name:", this.accountName)


    }

}

 // There is no authentication for this critical function

function createBankAccount(accNo,accType,accOwner,balance) {
   const user = new User();

   user.setAccountNumber(accNo);
   user.setAccountType(accType);
   user.setAccountOwnerName(accOwner);
   user.setBalance(balance);

    return user;
}


const user1 = createBankAccount("123","saving","Owner",300)

user1.getAccountDetail()